sklearn/impute/_base.py

```diff
@@ -426,6 +426,11 @@ def transform(self, X):
         ----------
         X : {array-like, sparse matrix}, shape (n_samples, n_features)
             The input data to complete.
+
+        Returns
+        -------
+        X_imputed : {ndarray, sparse matrix} of shape (n_samples, n_features)
```

Comment:
The number of features can be different from the input features:

1. If `add_indicator=True`, a missing value indicator will be added which can increase the number of features out.
2. Features that are all missing are removed by the imputer which can decrease the number of features out.

I suggest the following to use `n_features_out` instead:
```suggestion
        X_imputed : {ndarray, sparse matrix} of shape (n_samples, \
                    n_features_out)
```